modules/processors/nih_dataset_processor.py

```diff
@@ -0,0 +1,59 @@
+import datasets
+import os
+from ..dataset_processor import *
+
+class NIHDataset(Processor):
+
+    def __init__(self, config_name='number',is_query=True, *args, **kwargs):
+        self.dataset_name='nih_v1_'+config_name
+        if is_query:
+            self.dataset_name+='_query'
+        else:
+            self.dataset_name+='_doc'
+        self.conf_name= config_name
+        self.is_query=is_query
+        
+        super().__init__(*args, **kwargs, dataset_name=self.dataset_name)
+    
+    def process(self):
+        print(self.conf_name)
+        d = datasets.load_dataset('naver/bergen_nih_v1',self.conf_name)[self.split]
+        #oracle run data generations
+        qids_dids_l =[ (x['qid'],x['did']) for x in d]
+        #FIXME get the run dirs?
+        fname='runs/run.oracle.'+self.dataset_name+'.dev.trec'
+        #6915606477668963399	q0	10593264_2	0	100	run
+        f=open(fname,'w')
+        for i in qids_dids_l:
+            f.write(i[0]+'\tq0\t'+i[1]+'\t0\t100\trun\n')
+        f.close()
+        
+        if self.is_query:
+            #reprocess single str label into a list of str    
+            dataset = d.rename_column("qid", "id")
+            dataset = dataset.rename_column("query", "content")
+            dataset = dataset.remove_columns(['did', 'doc'])
+            dataset = dataset.map(lambda x: {"label": [str(x["label"])]})
+            print('QUERY:',dataset[:10])
+            return dataset
+        else:
+            #document dataset
+            dataset = d.rename_column("did", "id")
+            dataset = dataset.rename_column("doc", "content")
+            dataset = dataset.remove_columns(['qid', 'query','label'])
+            print(dataset[:10])
+            return dataset
+
```

Comment:
It would be cleaner to not subclass but add a different init_args in each different config: that's what configs are for :)

Reply:
That's what I did first, but then, the print_result function show the classname of the data_processor and not its arguments, leading to  several lines of NIHdataset making the results harder to read

Reply:
Ok, note for the future: use dataset name and not class name in print_results.